uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -656,13 +670,14 @@ define([
             initContact.call(self);
 
             if (self.isMode('edit')) {
-              initialLeaveRequestAttributes = self.request.attributes();
+              initialLeaveRequestAttributes = angular.copy(self.request.attributes());
+              // fileUploader property deleted because it will not be used
+              // in object comparison
+              delete initialLeaveRequestAttributes.fileUploader;
```

Comment:
Couple of points

> hasRequestChanged is a scope function,

but why? the function doesn't seem to be called from the view, and it is also used only in the parent controller, so why isn't it a simple private function?

> initialLeaveRequestAttributes is stored only once, so if I do the removal of property in hasRequestChanged then unnecessarily we have to remove the property every time 

I understand your point, but the `_.omit` being called multiple times (if it happens) has really a negligible effect, compared to the positive effect of having a better-structure code.

The point here is that you have a logic (object comparison) that is not contained in a isolated, reusable function, but rather it's spread both inside and outside the said function. Case in point is the fact that you need to leave a comment in `this.initAfterContactSelection` to explain what you are doing
```js
this.initAfterContactSelection = function () {
  // fileUploader property deleted because it will not be used
  // in object comparison
  initialLeaveRequestAttributes = _.omit(angular.copy(self.request.attributes()), 'fileUploader');
}
```
compare to how much more clean it is when you encapsulate the whole comparison logic in the `hasRequestChanged` function
```js
this.initAfterContactSelection = function () {
  initialLeaveRequestAttributes = angular.copy(self.request.attributes());
};

this.hasRequestChanged = function () {
  return !angular.equals(
    _.omit(initialLeaveRequestAttributes, 'fileUploader'),
    _.omit(updatedRequestAttributes, 'fileUploader')
  );
};
```
so unless there is a technical reason for why you simply cannot do the above, please change the code like that